sdk/trace/span.go

```diff
@@ -322,7 +322,7 @@ func startSpanInternal(tr *tracer, name string, parent core.SpanContext, remoteP
 		startTime = time.Now()
 	}
 	sk := o.SpanKind
-	if sk == "" {
+	if sk == apitrace.SpanKindUnspecified {
```

Comment:
Shouldn't we verify this in the same way you've done in the bridge mock?
```go
	switch spanKind {
	case oteltrace.SpanKindInternal,
		oteltrace.SpanKindServer,
		oteltrace.SpanKindClient,
		oteltrace.SpanKindProducer,
		oteltrace.SpanKindConsumer:
		// valid
	default:
		spanKind = oteltrace.SpanKindInternal
```

Maybe create a helper to get a valid SpanKind.